content/tutorials/build-apps/build-a-browser-wallet-in-js.md

```diff
@@ -0,0 +1,198 @@
+
+```json
+"scripts": {
+    "dev": "vite"
+}
+```
+
+If you get stuck while doing this tutorial, or working on another project, feel free to ask for help in the XRPL's [Developer Discord](https://discord.com/invite/KTNmhJDXqa).
+
+### Step 2: Creating the Home Page (Displaying Account & Ledger Details)
+
+In this step, we will create the home page that will display the account details and the ledger details.
+
+![Home Page Screenshot](img/js-wallet-home.png)
+
+1. Make a new folder named `src` in the root directory of the project.
+
+2. Create new files named `index.html`, `index.js` and `index.css`. 
+
+3. Copy the contents of the [index.html](_code-samples/build-a-wallet/js/index.html), [index.js](_code-samples/build-a-wallet/js/index.js) and [index.css](_code-samples/build-a-wallet/js/index.css) files to your files.
+
+Now that you've got the basic setup, let's discuss what we're going to do. For the home page, we want to:
+
+- Display our account info
+- Show what's happening on the ledger
+- And add a little logo for fun
+
+In order to make that happen, we need to look up our account state on the ledger. The next couple steps will walk you through that process.
+
+4. Make a new folder named helpers in the src directory, create a new file named `get-wallet-details.js` and copy the code written below to the file. In the function `getWalletDetails`, we are using the [account_info](https://xrpl.org/account_info.html) method to fetch the account details. We are also using the [server_info](https://xrpl.org/server_info.html) method to calculate the reserve base and increment. You can read more about the reserve requirement [here](https://xrpl.org/reserves.html).
+
+{{ include_code("_code-samples/build-a-wallet/js/src/helpers/get-wallet-details.js", language="js") }}
+
+5. Create another file in helpers folder named `get-ledger-details.js` which will be used to fetch the ledger details and copy the code written below to the file. In the function `getLedgerDetails`, we are using the [ledger](https://xrpl.org/ledger.html) that will fetch the details of the latest ledger.
+
+{{ include_code("_code-samples/build-a-wallet/js/src/helpers/get-ledger-details.js", language="js") }}
+
+6. Now, let's add the code to `index.js` file to fetch the account and ledger details and display them on the home page. Copy the code written below to the `index.js` file. Here we render the wallet and ledger details using the functions we implemented earlier in `get-wallet-details.js` and `get-ledger-details.js`. In order to make sure we have up to date data, we call the `getLedgerDetails` function every 4 seconds, as that's about how long it takes for a ledger to close on mainnet. 
+
+{{ include_code("_code-samples/build-a-wallet/js/index.js", language="js") }}
+
+The one other thing we do here is add two buttons - one to send XRP and one to view the transaction history of our account. They won't work just yet - we'll have to add those implementations next.
```

Comment:
Although I agree that going block by block is more descriptive, I find it a bit of an overhead because the developer now has to put all block pieces together, which takes a bit more time and also adds more jargon which can be summarised in one line comment.

But I am still open to change the code a bit if you or @JST5000 or @khancode suggest.